code/modules/modular_computers/file_system/programs/command/card.dm

```diff
@@ -0,0 +1,410 @@
+						for(var/jobtype in typesof(/datum/job))
+							var/datum/job/J = new jobtype
+							if(ckey(J.title) == ckey(t1))
+								jobdatum = J
+								break
+						if(!jobdatum)
+							to_chat(usr, "\red No log exists for this job: [t1]")
+							return
+
+						access = jobdatum.get_access()
+
+					modify.access = access
+					modify.assignment = t1
+					modify.rank = t1
+
+				callHook("reassign_employee", list(modify))
+
+		if("PRG_reg")
+			if(is_authenticated(usr))
+				var/temp_name = reject_bad_name(href_list["reg"])
+				if(temp_name)
+					modify.registered_name = temp_name
+				else
+					computer.visible_message("<span class='notice'>[src] buzzes rudely.</span>")
+
+		if("PRG_account")
+			if(is_authenticated(usr))
+				var/account_num = text2num(href_list["account"])
+				modify.associated_account_number = account_num
+
+		if("PRG_mode")
+			mode = text2num(href_list["mode_target"])
+
+		if("PRG_print")
+			if(!printing && computer)
+				printing = 1
+				playsound(computer.loc, 'sound/goonstation/machines/printer_dotmatrix.ogg', 50, 1)
+				spawn(50)
+					printing = 0
+					nanomanager.update_uis(src)
+					var/title
+					var/content
+					if(mode == 2)
+						title = "crew manifest ([worldtime2text()])"
+						content = "<h4>Crew Manifest</h4><br>[data_core ? data_core.get_manifest(0) : ""]"
+					else if(modify && !mode)
+						title = "access report"
+						content = {"<h4>Access Report</h4>
+							<u>Prepared By:</u> [scan && scan.registered_name ? scan.registered_name : "Unknown"]<br>
+							<u>For:</u> [modify.registered_name ? modify.registered_name : "Unregistered"]<br>
+							<hr>
+							<u>Assignment:</u> [modify.assignment]<br>
+							<u>Account Number:</u> #[modify.associated_account_number]<br>
+							<u>Blood Type:</u> [modify.blood_type]<br><br>
+							<u>Access:</u><div style="margin-left:1em">"}
+
+						var/first = 1
+						for(var/A in modify.access)
+							content += "[first ? "" : ", "][get_access_desc(A)]"
+							first = 0
+						content += "</div>"
+
+					if(content)
+						if(!printer.print_text(content, title))
+							to_chat(user, "<span class='notice'>Hardware error: Printer was unable to print the file. It may be out of paper.</span>")
+							return 1
+						else
+							computer.visible_message("<span class='notice'>\The [computer] prints out paper.</span>")
+
+		if("PRG_terminate")
+			if(is_authenticated(usr))
+				modify.assignment = "Terminated"
+				modify.access = list()
+
+				callHook("terminate_employee", list(modify))
+
+		if("PRG_make_job_available")
+			// MAKE ANOTHER JOB POSITION AVAILABLE FOR LATE JOINERS
+			if(is_authenticated(usr))
+				var/edit_job_target = href_list["job"]
+				var/datum/job/j = job_master.GetJob(edit_job_target)
+				if(!j)
+					return 1
+				if(can_open_job(j) != 1)
+					return 1
+				if(opened_positions[edit_job_target] >= 0)
+					time_last_changed_position = world.time / 10
+				j.total_positions++
+				opened_positions[edit_job_target]++
```

Comment:
Leaving stuff out in the open of `Topic` calls is kinda icky
but the precedent at this point has been set